### ISSUE TITLE
Exception thrown in Worker thread: GPU error: CUDA error in func set_header at line 165 invalid device symbol

### ISSUE THREAD

**Issue by javahaxxor** [ethereum-mining/ethminer#2115]
**Describe the bug**
I'm just starting out with ethminer and cannot run it with cuda 

**To Reproduce**
./ethminer -U -Z 1
ethminer 0.19.0-alpha.0
Build: linux/release/gnu

 i 23:09:54 ethminer Selected pool localhost:0
 i 23:09:54 ethminer Established connection to localhost:0
 i 23:09:54 ethminer Spinning up miners...
cu 23:09:54 cuda-0   Using Pci Id : 08:00.0 GeForce RTX 3060 Ti (Compute 8.6) Memory : 7.79 GB
 i 23:09:54 sim      Epoch : 0 Difficulty : 4.29 Gh
 i 23:09:54 sim      Job: 1cce3370… block 1 localhost:0
cu 23:09:55 cuda-0   Generating DAG + Light : 1.02 GB
cu 23:09:55 cuda-0   Unexpected error CUDA error in func set_constants at line 125 invalid device symbol on CUDA device 08:00.0
cu 23:09:55 cuda-0   Mining suspended ...
 m 23:09:59 ethminer 0:00 A0 0.00 h - cu0 0.00
 m 23:10:04 ethminer 0:00 A0 0.00 h - cu0 0.00
 m 23:10:09 ethminer 0:00 A0 0.00 h - cu0 0.00
--------------------------------------------------------------
./ethminer -U -M 0
ethminer 0.19.0-alpha.0
Build: linux/release/gnu

 i 23:12:02 ethminer Selected pool localhost:0
 i 23:12:02 ethminer Established connection to localhost:0
 i 23:12:02 ethminer Spinning up miners...
cu 23:12:02 cuda-0   Using Pci Id : 08:00.0 GeForce RTX 3060 Ti (Compute 8.6) Memory : 7.79 GB
 i 23:12:02 sim      Epoch : -1 Difficulty : 4.29 Gh
 i 23:12:02 sim      Job: 41b64834… block 0 localhost:0
 X 23:12:02 cuda-0   Exception thrown in Worker thread: GPU error: CUDA error in func set_header at line 165 invalid device symbol
 m 23:12:07 ethminer 0:00 A0 0.00 h - cu0 0.00
 m 23:12:12 ethminer 0:00 A0 0.00 h - cu0 0.00
--------------------------------------------------------------

**Expected behavior**

**Screenshots (Optional)**

**Environment (please complete the following information):**
- Operating System: [e.g. Windows 10]
- Hardware [e.g. GTX 1070]
- Ethminer Version [e.g. 0.14]
- Ethminer options used

**Additional context**

**Comment by javahaxxor** [comment]
@StefanOberhumer it looks like you might know why this happens, it's in ethash_cuda_miner_kernel.cu

**Comment by javahaxxor** [comment]
I could build it from source and it runs fine

**Comment by Gimel-ai** [comment]
You can check my compiled repo with RTX 30 series and cuda 11.1 support here - https://github.com/technopremium/ethminer_RTX30series

**Comment by jean-m-cyr** [comment]
NOTE: This above recommended repo violates the terms of GPL V3 licensing and is not recommended. It is a directly derived from Ethminer but publishes no source code as required.

Try this one instead: https://github.com/no-fee-ethereum-mining/nsfminer.git

**Comment by javahaxxor** [comment]
@Gimel-ai thanks but I completely agree with @jean-m-cyr. If you managed to add Cuda 11.1 support, congrats, and you should do a pull request to the main repo. I could then build it myself

**Comment by javahaxxor** [comment]
@jean-m-cyr what's the difference between nsfminer and ethminer?

**Comment by jean-m-cyr** [comment]
@javahaxxor Etheminer has been neglected due to lack of participation on the part of developers, testers, contributors in general. Over time it's garnered bugs and incompatibilities with newer driver/OS/hardware releases. My design philosophy favors clean, simple, and efficient implementation over tons of bells and whistles. That's were you'll see differences. Unused, frivolous, or just plain nonsense parameters are gone, some of them replaced with automation. The console log is de-cluttered by removing most non mining pertinent info, the time-of-day on every line for example.

In short: accumulated bug fixes, improved mining algorithm, a project not blocked by lack of participation but that remains open sourced.

**Comment by javahaxxor** [comment]
Cool. I'd love to try it and start contributing

**Comment by ShawnLau11** [comment]
> I could build it from source and it runs fine

Hi @javahaxxor, I'm facing the same issue with the exe i build from source. May I know which commit u build from?

**Comment by javahaxxor** [comment]
Oh, but that was a long time ago.. I've since then found the compiled binary 😁. Sorry, I can't remember what I did to get it to work

**Comment by encoderlee** [comment]
![image](https://user-images.githubusercontent.com/9878012/146749463-a069f400-4434-461f-8146-49761f70007d.png)
insert
list(APPEND CUDA_NVCC_FLAGS "-gencode arch=compute_86,code=sm_86")
into file
libethash-cuda/CMakeLists.txt
then rebuild to support RTX30xx gpu

**Comment by fadyanwar** [comment]
@encoderlee Hi, thanks for the advice, would this work with Nvidia Tesla V100? Also, I'm following the steps to build it but it's failing with the below error. Could be an outdated package. So I was wondering if someone had forked and corrected this on another repo. Thanks.


gmake[2]: *** [CMakeFiles/Boost.dir/build.make:99: Boost-prefix/src/Boost-stamp/Boost-download] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/Boost.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

[hunter ** FATAL ERROR **] Build step failed (dir: /home/fady/.hunter/_Base/4b894e1/a5c138f/cb395ce/Build/Boost
[hunter ** FATAL ERROR **] [Directory:/home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/projects/Boost]

------------------------------ ERROR -----------------------------
    https://docs.hunter.sh/en/latest/reference/errors/error.external.build.failed.html
------------------------------------------------------------------

CMake Error at /home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/modules/hunter_error_page.cmake:12 (message):
Call Stack (most recent call first):
  /home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/modules/hunter_fatal_error.cmake:20 (hunter_error_page)
  /home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/modules/hunter_download.cmake:614 (hunter_fatal_error)
  /home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/projects/Boost/hunter.cmake:381 (hunter_download)
  /home/fady/.hunter/_Base/Download/Hunter/0.23.112/4b894e1/Unpacked/cmake/modules/hunter_add_package.cmake:62 (include)
  CMakeLists.txt:64 (hunter_add_package)


-- Configuring incomplete, errors occurred!
See also "/home/fady/ethminer/build/CMakeFiles/CMakeOutput.log".

**Comment by fadyanwar** [comment]
Answering my own question, yes it does, as shown below. Thanks @encoderlee  


nvidia-smi
Sun Jan 23 15:51:40 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 495.46       Driver Version: 495.46       CUDA Version: 11.5     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:1E.0 Off |                    0 |
| N/A   56C    P0   205W / 300W |  10307MiB / 16160MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       752      G   /usr/lib/xorg/Xorg                  4MiB |
|    0   N/A  N/A      1715      C   ./ethminer                      10300MiB |
+-----------------------------------------------------------------------------+


I've forked this repo and added the above mentioned fix as well as updating build doc with a workaround Boost dependency download timeout. You can find my fork here https://github.com/fadyanwar/ethminer/blob/1a713e4d879bcc4187f2bce66330a52e050018a0/docs/BUILD.md
